blueprint/cli/src/bin/db.rs

```diff
@@ -65,105 +67,131 @@ async fn cli() {
+                    ui.indent();
+                    match migrate(&mut ui, &config.database).await {
+                        Ok(migrations) => {
+                            ui.outdent();
+                            ui.success(&format!("{} migrations applied.", migrations));
+                            Ok(())
+                        }
+                        Err(e) => {
+                            ui.outdent();
+                            ui.error("Could not migrate database!", &e);
+                            Err(e)
+                        }
                     }
                 }
-            }
-            Commands::Seed => {
-                ui.info(&format!("Seeding {} database…", &cli.env));
-                match seed(&config.database).await {
-                    Ok(_) => ui.success("Seeded database successfully."),
-                    Err(e) => ui.error("Could not seed database!", e),
-                }
-            }
-            Commands::Reset => {
-                ui.info(&format!("Resetting {} database…", &cli.env));
-                ui.indent();
-                match reset(&mut ui, &config.database).await {
-                    Ok(db_name) => {
-                        ui.outdent();
-                        ui.success(&format!("Reset database {} successfully.", db_name));
-                    }
-                    Err(e) => {
-                        ui.outdent();
-                        ui.error("Could not reset database!", e)
+                Commands::Seed => {
+                    ui.info(&format!("Seeding {} database…", &cli.env));
+                    match seed(&config.database).await {
+                        Ok(_) => {
+                            ui.success("Seeded database successfully.");
+                            Ok(())
+                        }
+                        Err(e) => {
+                            ui.error("Could not seed database!", &e);
+                            Err(e)
+                        }
                     }
                 }
-            }
-            Commands::Prepare => {
-                if let Err(e) = ensure_sqlx_cli_installed(&mut ui).await {
-                    ui.error("Error ensuring sqlx-cli is installed!", e);
-                    return;
+                Commands::Reset => {
+                    ui.info(&format!("Resetting {} database…", &cli.env));
+                    ui.indent();
+                    match reset(&mut ui, &config.database).await {
+                        Ok(db_name) => {
+                            ui.outdent();
+                            ui.success(&format!("Reset database {} successfully.", db_name));
+                            Ok(())
+                        }
+                        Err(e) => {
+                            ui.outdent();
+                            ui.error("Could not reset database!", &e);
+                            Err(e)
+                        }
+                    }
```

Comment:
I'm not super fond of the pattern here, where we log the error manually in every single place.
I think it'd be more robust to do this logging in the `main` function above, before exiting with a non-zero code.

I imagine this was originally done to control the message shown to the user. We can achieve the same goal with the following structure:

```suggestion
                    let outcome = reset(&mut ui, &config.database).await.context("Could not reset the database!");
                    ui.outdent();
                    let db_name = outcome?; 
                    ui.success(&format!("Reset database {} successfully.", db_name));
                    Ok(())
```

and in `main`:

```rust
 match cli().await {
        Ok(_) => ExitCode::SUCCESS,
        Err(e) => {
            ui.error(e.to_string(), &e);
            ExitCode::FAILURE,
        }
    }
```

Reply:
Ideally `ui.error` would just take the error, relying on its `Display` implementation, rather than forcing the caller to invoke `e.to_string()`, but that can be fixed later.

Reply:
fixed